src/Compilers/CSharp/Portable/Symbols/Source/ThisParameterSymbol.cs

```diff
@@ -179,7 +179,8 @@ internal override ScopedKind EffectiveScope
             {
                 var scope = _containingType.IsStructType() ? ScopedKind.ScopedRef : ScopedKind.None;
                 if (scope != ScopedKind.None &&
-                    HasUnscopedRefAttribute)
+                    HasUnscopedRefAttribute &&
+                    UseUpdatedEscapeRules)
```

Comment:
I am seeing this pair of checks very frequently in this PR. I am wondering when we would *not* want to do this check, i.e. when would we behave as if `HasUnscopedRefAttribute` is true, even when old escape rules are being used.

Essentially, should we replace `HasUnscopedRefAttribute` itself with some similar helper which already implies that new escape rules are being used.

Reply:
> when would we behave as if `HasUnscopedRefAttribute` is true, even when old escape rules are being used.

We still look at `HasUnscopedRefAttribute` without checking `UseUpdatedEscapeRules` when synthesizing delegates for example - the synthesized delegate might have `[UnscopedRef]` even if it's synthesized in an old-rules assembly - if it points to a method from an updated-rules assembly. See tests UnscopedRefAttribute_RefSafetyRules_Reference_SynthesizedDelegate_*.